types/types.go

```diff
@@ -23,67 +24,177 @@ import (
 	"github.com/prometheus/common/model"
 )
 
+type State uint8
+
+const (
+	Unprocessed State = iota
+	Active
+	Silenced
+	Inhibited
+)
+
+// AlertStatus stores the state and values associated with an Alert.
+type AlertStatus struct {
+	status State
+	values []string
+}
+
+// Save an allocation when serializing an empty string slice.
+var emptyVals = []string{}
+
+// MarshalJSON implements the json.Marshaler interface.
+func (a *AlertStatus) MarshalJSON() ([]byte, error) {
+	values := a.values
+	if values == nil {
+		values = emptyVals
+	}
+	return json.Marshal(map[string]interface{}{
+		"status": a.status,
+		"values": values,
+	})
+}
+
+// Save an allocation when serializing the unknown response.
+const unknown = "unknown"
+
+// MarshalJSON implements the json.Marshaler interface.
+func (s State) MarshalJSON() ([]byte, error) {
+	status, found := statusMap[s]
+	if !found {
+		status = unknown
+	}
+
+	return json.Marshal(status)
+}
+
+var statusMap = map[State]string{
+	Unprocessed: "unprocessed",
+	Active:      "active",
+	Silenced:    "silenced",
+	Inhibited:   "inhibited",
```

Comment:
An alert can be silenced and inhibited. More so once we move towards a model where this is shown immediately and not just after initial processing.

So "unprocessed", "active", and "suppressed" might be more accurate? @brian-brazil